### PR TITLE
Adds Chess Pieces

### PR DESCRIPTION
**Pull request by Mitchs98**
## What Does This PR Do
This adds the chess pieces from the Virtual Gameboard Multiplayer thread(seen here; https://www.paradisestation.org/forum/topic/22211-virtual-gameboard-multiplayer/) as well as the icons for the floor decals. Though I did not add in the floor decals to the floor painter or such as I'm not very comfortable messing with those. Full credit for these sprites goes to Slime King Dana AKA Dan Jello. Each piece is craftable with one cardboard to make things nice and easy to obtain.

It could also be added to the merch console if desired.

## Why It's Good For The Game
Players have wanted multiplayer chess and other games for a while now. I understand that, while it is theoretically possible to get the gameboard working for two people, it would require a lot of extra effort and would be a lot harder to implement. Thus it was suggested to simply add chess pieces craftable with cardboard. As a big fan of this and as someone who constructs large chessboards out of random items it was natural I wanted to implement this idea.

On a whole it will allow people something to do, another project to build, and be a whole lot easier than chess with some shoes, masks, jumpsuits, and food items. 

## Images of changes
dreamseeker_n1sXvYaOco

## Changelog
:cl:
add: Added craftable Chess pieces to allow players to construct chess boards.
/:cl:

### CODE REVIEW

**Comment by McRamon** [comment]
having playable chess is nice but this is far from realisation i personally want to see, Chess shouldn't take over an entire room.

**Comment by Mitchs98** [comment]
It doesn't take over an entire room. It takes an 8x8 space. I'm unsure why that's an issue, people do it already just with random junk.

File: code/game/objects/items/chess.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
+//Cardboard chess pieces. To be used for multiplayer chessboards. An alternative to trying to get a coded game-board set up.
+
+/obj/item/chesspiece/
```
Comment (lewcc) [inline code/game/objects/items/chess.dm:3]:
```suggestion
/obj/item/chesspiece
```

**Comment by Carthusia** [comment]
> having playable chess is nice but this is far from realisation i personally want to see, Chess shouldn't take over an entire room.

To be fair, while i agree about the size thing and would love for the chess game to work in MP, people do make large builds already with all kinds of wacky concepts, a big chess room isnt to far from what people already do. IE: Giant lasertag duel arenas/Plasma/Nitrogen rooms etc.

Its just one more possible thing for people to make every once and awhile. 

OH, perhaps also add tiny versions as well similar to the little talking toys around.

Maybe even add a new table crafting type that you build a table that has the surface texture of a chessboard, say metal table frame a black/white carpet. This way people can use the tiny versions on the table sized chessboard, AND people can tableflip the thing when their loosing.

**Comment by atteria** [comment]
This is a good addition to support what people already do when they want to play chess in-game, in lieu of an actual multiplayer gameboard.

**Comment by matttheficus** [comment]
Honestly would prefer a remake of the chess table than this. Wouldn't need the logic, simply an 8x8 grid for players to use as anyone who would want to play chess ICly likely already knows the rules.

**Comment by LightFire53** [comment]
An interesting alternative could be to make this a holodeck thing, which would then keep the large sized stuff while keeping it in a "designated zone"?

**Comment by Carthusia** [comment]
> An interesting alternative could be to make this a holodeck thing, which would then keep the large sized stuff while keeping it in a "designated zone"?

ooohhhh thats a nice one, perhaps both that and the tiny chess pieces to with a buildable table textured like a chessboard

**Comment by Mitchs98** [comment]
> An interesting alternative could be to make this a holodeck thing, which would then keep the large sized stuff while keeping it in a "designated zone"?

That _could_ be an alternative. I don't believe the holodeck is quite big enough to house the chessboard though? A proper board needs an 8x8 area.

**Comment by Mitchs98** [comment]
> Maybe even add a new table crafting type that you build a table that has the surface texture of a chessboard, say metal table frame a black/white carpet. This way people can use the tiny versions on the table sized chessboard, AND people can tableflip the thing when their loosing.

My only reservations with this are the difficulty in scaling the pieces correctly to where you can see what they are and move them easily on the smaller table as well as how hard it would be to properly click on squares. I would be willing to try it instead if someone sprites it, I personally can't sprite anything at all :D

**Comment by shmabuu** [comment]
Really like this. Though, while I don't at all expect you to do this, would it be possible to work this into the holodeck somehow? Even if it needs some expanding, I'm fairly certain at least one other server has done chess with the holodeck and would give more use to it as it's not too often the holodeck gets used.

**Comment by Christasmurf** [comment]
> > An interesting alternative could be to make this a holodeck thing, which would then keep the large sized stuff while keeping it in a "designated zone"?
> 
> That _could_ be an alternative. I don't believe the holodeck is quite big enough to house the chessboard though? A proper board needs an 8x8 area.

I think what they mean by this suggestion is like
So the chessboard machine right? You have an 8x8 area on admin zlevel, where the holodeck templates are
The players connect to this area with it like the uh, the xenobio console thing. You can use it to grab and place the pieces where you want them. So there's no ACTUAL physical place you can go into as a crewmember, but instead it's like an off-station room that connects via the holochess console.

Unsure how difficult it would be to make that work, but that's what I think they mean by that suggestion

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 2]
```diff
@@ -341,7 +341,19 @@ GLOBAL_LIST_INIT(cardboard_recipes, list (
 	new /datum/stack_recipe("folder", /obj/item/folder),
 	new /datum/stack_recipe("cardboard tube", /obj/item/c_tube),
 	new /datum/stack_recipe("cardboard box", /obj/structure/closet/cardboard, 4),
-	new /datum/stack_recipe("cardboard cutout", /obj/item/cardboard_cutout, 5)
+	new /datum/stack_recipe("cardboard cutout", /obj/item/cardboard_cutout, 5),
+	new /datum/stack_recipe("black pawn", /obj/item/chesspiece/bpawn, 1),
```
Comment (dearmochi) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:345]:
```suggestion
	null,
	new /datum/stack_recipe("black pawn", /obj/item/chesspiece/bpawn, 1),
```

Add a `null` to separate the above stuff with the chess thingies

**Review by dearmochi — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 2 above)
Comment (Mitchs98) [reply code/game/objects/items/stacks/sheets/sheet_types.dm:346]: Done~

**Review by dearmochi — APPROVED** [review]

**Pull request merged**